src/platform/linux/x11/window.rs

```diff
@@ -380,6 +380,35 @@ impl UnownedWindow {
                 window.set_always_on_top_inner(window_attrs.always_on_top).queue();
             }
 
+            if window_attrs.center_window && !window_attrs.fullscreen.is_some() 
+            {
+                let window_size = window.get_outer_size();
+
+                let monitor;
+                if window_attrs.start_monitor == -1 {
```

Comment:
As mentioned, this code would look better if you just stored a `MonitorId` here.